examples/energy-management-app/energy-management-common/include/EnergyEvseTargetsStore.h

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/energy-evse-server/energy-evse-server.h>
+#include <credentials/FabricTable.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/Pool.h>
+
+#include <app-common/zap-generated/cluster-objects.h>
+
+#include <ChargingTargetsMemMgr.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace EnergyEvse {
+
+class EvseTargetsDelegate : public chip::FabricTable::Delegate
+{
+public:
+    EvseTargetsDelegate();
+    ~EvseTargetsDelegate();
+
+    CHIP_ERROR Init(PersistentStorageDelegate * targetStore);
+
+    /**
+     * @brief Delegate should implement a handler for LoadTargets
+     *
+     * This needs to load any stored targets into memory
+     */
+    CHIP_ERROR LoadTargets();
+
+    /**
+     *  @brief   This returns a reference to the existing targets
+     */
+    const DataModel::List<const Structs::ChargingTargetScheduleStruct::Type> & GetTargets();
+
+    /**
+     * @brief   Copies a ChargingTargetSchedule into our store
+     *
+     * @param [in] an entry from the SetTargets list containing:
+     *             dayOfWeekForSequence and chargingTargets (list)
+     *
+     * This routine scans the existing targets to see if we have a day of week
+     * set that matches the new target dayOfWeek bits. If there is an existing
+     * matching day then it replaces the days existing targets with the new entry
+     */
+    CHIP_ERROR SetTargets(
+        const DataModel::DecodableList<Structs::ChargingTargetScheduleStruct::DecodableType> & chargingTargetSchedulesChanges);
+
+    /**
+     *  @brief   This deletes all targets and resets the list to empty
+     */
+    CHIP_ERROR ClearTargets();
+
+    /**
+     * Part of the FabricTable::Delegate interface. Gets called when a fabric is deleted, such as on FabricTable::Delete().
+     **/
+    virtual void OnFabricRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex) override;
+
+private:
+    static uint16_t GetTlvSizeEstimate();
```

Comment:
Estimate seems like a `guess` in this context. This need some comment explaining this - is this an upper bound? 

We seem to use this to allocate memory to load things. In that case it does sound like we allocate sufficient size to be able to load anything. Is this correct? Should be this named accordingly (like upper bound/max-size/loading-size or something like that)?

Reply:
GetTlvSizeEstimate renamed to GetTlvSizeUpperBound